compiler/rustc_data_structures/src/graph/dominators/mod.rs

```diff
@@ -287,14 +287,14 @@ impl<Node: Idx> Dominators<Node> {
     }
 
     pub fn is_dominated_by(&self, node: Node, dom: Node) -> bool {
-        // FIXME -- could be optimized by using post-order-rank
```

Comment:
I'm not sure this should be removed? It seems like we could still potentially optimize the loop here to avoid scanning all the dominators. I'd need to refresh my memory on the details to say how, but it doesn't seem like the change to an unrelated piece of code in middle/mir will fix this.

Reply:
I think it may be that if the post-order-rand of `node` is greater than `dom`'s during the scan process, it can be interrupted and return false? I think it's worth a try, but may add more extra calculations

Reply:
That makes sense! I will modify the code

Reply:
That makes sense! In this case we don't need to make any changes in `rustc_middle`